### ISSUE TITLE
Add `@macro` comment expansion

### ISSUE THREAD

**Issue by feinstein** [flutter/flutter-intellij#7203]
Many times we are looking at some widget's source-code and we find something like this:

```dart
  /// {@macro flutter.services.TextInputConfiguration.enableIMEPersonalizedLearning}
  final bool enableIMEPersonalizedLearning;
```

I can go to the variable and hit the shortcut for showing the definition, which in turn will pull the macro text, but I wanted to hit F4 and go to the macro definition itself, so I can read it as code, and not in the small doc pop-up window.

**Comment by helin24** [comment]
@DanTup does this go to definition of dartdoc macro exist for VS Code? I wonder if this would need changes in the analysis server.

**Comment by DanTup** [comment]
No, we don't support this in VS Code - but I think we probably should. Probably we should support both of these?

1. Ctrl+Click on the name (`flutter.services.TextInputConfiguration.enableIMEPersonalizedLearning`) goes to where the template is defined
2. Hovering over the name shows the resolved text (you can hover over the variable `enableIMEPersonalizedLearning` already, but I don't see why we shouldn't also support it on the name in the comment)

I'm not sure if server currently tracks the locations of the templates though (I think it might only keep the string content), so that might need to change.

If there's nothing IntelliJ-specific if the server handled these, then perhaps this issue can just be moved to the SDK? (although I'm not sure if you can move across orgs?)

**Comment by helin24** [comment]
I'm not sure about what specific IntelliJ work would be required, so I'll keep this open. I'm not able to move to the Dart SDK repo but I can copy to there.
